src/Reflection/ReflectionClass.php

```diff
@@ -338,20 +338,33 @@ private function getMethodsIndexedByName() : array
     /**
      * Fetch an array of all methods for this class.
      *
+     * @param ?int $filter
      * @return ReflectionMethod[]
      */
-    public function getMethods() : array
+    public function getMethods(?int $filter = null) : array
     {
-        return array_values($this->getMethodsIndexedByName());
+        if (null === $filter) {
```

Comment:
Yes, well aware of the optimisation - still a huge load of noise for nothing IMO